apps/src/netsim/NetSimSendWidget.js

```diff
@@ -54,24 +86,123 @@ NetSimSendWidget.createWithin = function (element, connection) {
 NetSimSendWidget.prototype.bindElements_ = function () {
   this.rootDiv_ = $('#netsim_send_widget');
   this.toAddressTextbox_ = this.rootDiv_.find('#to_address');
-  this.payloadTextbox_ = this.rootDiv_.find('#payload');
+  this.toAddressTextbox_.bind('keyPress', function (event) {
+    // Don't block control combinations (copy, paste, etc);
+    if (event.metaKey || event.ctrlKey) {
+      return true;
+    }
+
+    // Prevent non-digit characters
+    var charCode = !event.charCode ? event.which : event.charCode;
```

Comment:
Are we still trying to transition to using jQuery in /apps?  If so I think they abstract away some of the charCode weirdness between different browsers.